### DISCUSSION TITLE
How I can create my personal fixture?

### DISCUSSION THREAD

**Discussion by lozik4** [seleniumbase/SeleniumBase#1631] [Q&A] [answered]
I use conftest.py about save my pytest fixture. How do it in the Selenium base?

**Comment by mdmintz** [comment, marked as answer]
You can still use a ``conftest.py`` file with SeleniumBase. Put it in the root folder when you run your tests.
If you're trying to modify the existing ``sb`` fixture, you can use the example from here: https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md#sb_sf_10
(Eg. https://github.com/seleniumbase/SeleniumBase/blob/master/examples/test_override_sb_fixture.py)

**Reply by lozik4** [reply]
Page class
```
from seleniumbase import BaseCase

class IndexPage(BaseCase):

    def open_start_page(self):
        self.open('url')

    def open_mobile_page(self):
        self.click_xpath("//a/span[text()='Mobile app']")
```

Test Class
```
from pages.index_page import IndexPage

class TestIndexPage(IndexPage):

    def test_mobile_app(self, fake_my):

        self.open_start_page()

        print(fake_my)

        self.open_mobile_page()

```

conftest.py
```
import pytest

@pytest.fixture()
def fake_my():
    return 123456
```


If I run test
```
    def _callTestMethod(self, method):
>       method()
E       TypeError: test_mobile_app() missing 1 required positional argument: 'fake_my'
```

**Reply by mdmintz** [reply]
If using your own fixture, you'll need to use that with the seleniumbase ``sb`` fixture format:
https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md#sb_sf_03 or https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md#sb_sf_04

**Comment by lozik4** [comment]
If I use pytest and write my custom fixture in Page Class I must inheritance BaseCase?

**Reply by mdmintz** [reply]
If using your own fixture, you'll need to use that with the seleniumbase ``sb`` fixture format:
https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md#sb_sf_03 or https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md#sb_sf_04

**Reply by lozik4** [reply]
Thanx!
